OTAnalytics/plugin_ui/customtkinter_gui/dummy_viewmodel.py

```diff
@@ -769,13 +815,31 @@ def _draw_arrow_for_selected_flows(self) -> None:
         for flow in self._get_selected_flows():
             if start_section := self._application.get_section_for(flow.start):
                 if end_section := self._application.get_section_for(flow.end):
+                    start_refpt_calculator = self._get_section_refpt_calculator(
+                        start_section
+                    )
+                    end_refpt_calculator = self._get_section_refpt_calculator(
+                        end_section
+                    )
                     ArrowPainter(self._canvas).draw(
                         start_section=start_section,
                         end_section=end_section,
+                        start_refpt_calculator=start_refpt_calculator,
+                        end_refpt_calculator=end_refpt_calculator,
                         tags=[LINE_SECTION],
                         arrow_style=ARROW_STYLE,
                     )
 
+    def _get_section_refpt_calculator(
+        self, section: Section
+    ) -> SectionRefPointCalculator:
+        if isinstance(section, LineSection):
+            return InnerSegmentsCenterCalculator()
+        elif isinstance(section, Area):
+            return GeometricCenterCalculator()
+        else:
+            raise ValueError("section has to be a LineSection or an Area, but isnt")
```

Comment:
See comment in line 674

Reply:
We could also do something like
```python
calculator_map: dict[Type, SectionRefPointCalculator] = {
      LineSection, InnerSegmentsCenterCalculator(),
      Area, GemetricCenterCalculator(),
      }
        dummy.get(type(section))
```

The dictionary could be setup in the `__init__` method. This would allow to add future types by adding a new entry into the dictionary instead of changing the code. (See Open Closed Principle for more about that: https://de.wikipedia.org/wiki/Open-Closed-Prinzip )